### PR TITLE
Fix compiling assets and update deploy strategy

### PR DESCRIPTION
**Pull request by girardinsamuel**
- [x] fixed npm run watch infinite reload (due to tailwind config js which was checking compiled js/css files triggering an infinite loop)
- [x] updated contributing doc
- [x] removed compiled assets from source control 
- [x] bumped laravel-mix version 

For exceptionite, I was thinking that the easiest way to handle compiled assets was to remove them from source control. This way when opening a PR, we don't have issue with merging compiled assets or compiled assets not up to date. And when releasing, the deploy job on github ci would compile assets, create the package an upload it on pypi (current behaviour). But then the tip of master branch which correspond to the release do not have compiled assets in it, so for example installing package from git repo would not work.

- [x] So the solution would be that: the deploy workflow could create a new commit with the compiled assets. This way last tagged version of master is the same as the pypi package.

### CODE REVIEW

**Comment by josephmancuso** [comment]
Needs more thought before merging

**Comment by girardinsamuel** [comment]
@josephmancuso I update the workflow and tried it by creating a release on my forked repo on a specific branch, it looks like it's working !

So TLDR:
- when developing in local you can run `npm run watch` or `npm run production` to compile assets and it won't appear as git changes (as ignored, but tracked).
- when opening a PR, compiled assets are contained in the PR, tests are passing
- when merging a PR to master or pushing to master, nothing particular happens. Tests are run with latest version of assets
- when creating a release:
  - the release will tag the current tip of master with v.X.Y.Z
  - the deploy job will be triggered
  - it will compile assets
  - it will run tests with latest assets
  - it will add those compile assets (by forcing because ignored) to the repo and create a new commit
  - it will push the commit to master (automatically) and move the v.X.Y.Z tag at the tip of the branch
  - it will publish the release

You end up with a release vX.Y.Z on PyPi with assets compiled automatically, and a tagged commit corresponding to the github releases with the compiled assets up to date for this commit.
If someone install from git he will get latest compiled assets from the latest release
